src/library.tests/WorkloadInformationProviderTests.cs

```diff
@@ -47,63 +47,97 @@ public async void GetReachableServicesAsync_HeadlessService(int numServices, int
-                                Addresses =  new List<V1EndpointAddress>()
-                            }
-                        };
-                if (!addSubset) {
-                    subsets = null;
-                }
-                var endPoint = new V1Endpoints()
-                {
-                    Subsets = subsets,
-                    Metadata = new V1ObjectMeta()
-                    {
-                        Name = $"{namingFunction(i)}"
-                    }
-                };
-                if (addSubset) {
-                    for (int j = 0; j < numAddresses; j++)
-                    {
-                        endPoint.Subsets[0].Addresses.Add(new V1EndpointAddress
-                        {
-                            Hostname = addressHostNamingFunction(j)
-                        });
-                    }
-                }
-                else {
-                    // we only want to skip addign subset in one, the rest should have subsets
-                    addSubset = true;
+                    Assert.Equal(endpoint.Ports.ElementAt(0).LocalPort, -1);
+                    Assert.Empty(endpoint.Ports.Where(p => portsToIgnore.Contains(p.RemotePort)));
                 }
-
-                A.CallTo(() => _autoFake.Resolve<IKubernetesClient>().GetEndpointInNamespaceAsync(namingFunction(i), A<string>._, A<CancellationToken>._)).Returns(endPoint);
             }
-            A.CallTo(() => _autoFake.Resolve<IKubernetesClient>().ListServicesInNamespaceAsync(default, default, default)).WithAnyArguments().Returns(new V1ServiceList(serviceList));
         }
-    }
+
+        private void ConfigureHeadlessService(int numServices, Func<int, string> namingFunction, int numAddresses, Func<int, string> addressHostNamingFunction, string ignorePorts = null)
+        {
+            var serviceList = new List<V1Service>();
+            // introducing this variable so we can add an endpoint with empty subset to have crash coverage
+            bool addSubset = false;
+            for (int i = 0; i < numServices; i++)
+            {
+                var service = new V1Service()
+                {
+                    Spec = new V1ServiceSpec()
+                    {
+                        Type = "ClusterIP",
+                        ClusterIP = "None",
+                        Ports = new List<V1ServicePort> { new V1ServicePort(port: 80, protocol: "TCP") },
+                        Selector = new Dictionary<string, string> { { "app", "myapp" } }
+                    },
+                    Metadata = new V1ObjectMeta()
+                    {
+                        Name = namingFunction(i),
+                    }
+                };
+
+                if (!string.IsNullOrEmpty(ignorePorts))
+                {
+                    service.Metadata.Annotations = new Dictionary<string, string>();
+                    service.Metadata.Annotations.Add("bridgetokubernetes/ignore-ports", ignorePorts);
```

Comment:
Aside from when ignore-ports is present with valid value, lets add coverage for when ignore-ports is empty, not present at all, or present with bad formatted string